src/System.Net.Sockets/src/System/Net/Sockets/SimpleContextAwareResult.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Net
+{
+    // This class ensures that the user callback runs in the caller's ExecutionContext.
+    internal partial class SimpleContextAwareResult : LazyAsyncResult
```

Comment:
The old class still exists in Common, and is used by a couple projects.  Switching those would be trickier because they rely on some additional functionality in ContextAwareResult.